### PR TITLE
Upgraded all tools to rubocop 0.38.0

### PR DESCRIPTION
**Pull request by KrauseFx**
This will now finally work together with the hound! :rocket:

### CODE REVIEW

**Comment by KrauseFx** [comment]
The Hound is happy, good good

File: credentials_manager/lib/credentials_manager/appfile_config.rb [hunk 1]
```diff
@@ -144,7 +144,7 @@ def for_lane(lane_name, &block)
     #             be applied.
     def for_platform(platform_name, &block)
       if ENV["FASTLANE_PLATFORM_NAME"] == platform_name.to_s
-        block.call
+        yield
```
Comment (mfurtak) [inline credentials_manager/lib/credentials_manager/appfile_config.rb]: Do we want to remove the `&block` params if they are now unused?

Comment (KrauseFx) [inline credentials_manager/lib/credentials_manager/appfile_config.rb]: It's still a parameter, right?

File: fastlane_core/lib/fastlane_core/configuration/configuration_file.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@ def method_missing(method_sym, *arguments, &block)
         return unless self.config._values[method_sym].to_s.empty?
 
         value = arguments.first
-        value = block.call if value.nil? && block_given?
+        value = yield if value.nil? && block_given?
```
Comment (mfurtak) [inline fastlane_core/lib/fastlane_core/configuration/configuration_file.rb]: remove `&block` param if unused?

File: fastlane/lib/fastlane/fast_file.rb [hunk 3]
```diff
@@ -105,7 +105,7 @@ def platform(platform_name, &block)
 
       self.current_platform = platform_name
 
-      block.call
+      yield
```
Comment (mfurtak) [inline fastlane/lib/fastlane/fast_file.rb]: remove `&block` param if unused?

File: credentials_manager/lib/credentials_manager/appfile_config.rb (hunk 1 above)
Comment (mfurtak) [inline credentials_manager/lib/credentials_manager/appfile_config.rb]: I believe it doesn't need to be stated unless you want to interact with it directly (like `block.call`). Any Ruby method can `yield` to a block even if no block parameter is stated.

**Comment by mfurtak** [comment]
Looks good to me, with questions about possibly removing unused `&block` params. :+1:

**Comment by mfurtak** [comment]
Looks like `spaceship` had a couple things not pass :disappointed:

**Comment by KrauseFx** [comment]
@mfurtak what would you do about the block parameter? Just remove it? Would `yield` still work then?

**Comment by KrauseFx** [comment]
Updated the PR - the spaceship issue was because the `spaceship.gemspec` was still pointing to the old rubocop version

**Comment by mfurtak** [comment]
Yep, `yield` will still work without the `&block` parameter

**Comment by mfurtak** [comment]
@KrauseFx Good to go?

**Comment by KrauseFx** [comment]
Yeah, removing the parameters now

**Comment by mfurtak** [comment]
:rocket: :sparkles:

**Comment by KrauseFx** [comment]
Build failing, let's fix this :+1:

File: fastlane/spec/actions_specs/create_keychain_spec.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@
         expect(result[1]).to_not include '-u'
         expect(result[1]).to include '~/Library/Keychains/test.keychain'
         expect(result[2]).to start_with "security list-keychains -s"
-        expect(result[2]).to end_with "#{File.expand_path('~/Library/Keychains/test.keychain')}"
+        expect(result[2]).to end_with File.expand_path('~/Library/Keychains/test.keychain').to_s
```
Comment (lacostej) [inline fastlane/spec/actions_specs/create_keychain_spec.rb]: is this to_s necessary ?

Comment (KrauseFx) [inline fastlane/spec/actions_specs/create_keychain_spec.rb]: Good point, it's not

**Pull request merged**